test/fast/Unit tests/nvm_ensure_posix_path

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+cleanup () {
+  unset -f die cleanup
+  unset NVM_SOURCE out
+  # sudo rmdir '/mnt/z' 2>/dev/null
+  # sudo rmdir '/mnt' 2>/dev/null
+  # sudo rmdir '/drives/z' 2>/dev/null
+  # sudo rmdir '/drives/w' 2>/dev/null
+  # sudo rmdir '/drives' 2>/dev/null
+}
+die () { echo "$@" ; cleanup ; exit 1; }
+
+\. ../../../nvm.sh
+
+set -ex
+
+# sudo mkdir -p '/mnt/z'
+# sudo mkdir -p '/drives/z'
+# sudo mkdir -p '/drives/w'
```

Comment:
It would be needed to create those folders to simulate Windows and properly unit test the function...

Reply:
it seems reasonable, some of the tests already need sudo. we just need to make sure they don’t exist already, and should error out if they do to avoid clobbering user dirs.

Reply:
true, or check to create and remove them only if they dont already exist.

Reply:
not needed anymore